### PR TITLE
Allow for adjustement of the metric control geometry 

### PR DESCRIPTION
**Pull request by namnguyen20999**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Added logic to track changes in the element’s bounding box dimensions to verify if the element has grown or shrunk after resizing operations.

## Checklist
_We encourage you to keep the code coverage percentage at 80% and above._

- [X] Does this solution meet the acceptance criteria of the related issue?
- [ ] Is the related issue checklist completed?
- [ ] Does this PR adds unit tests for the developed code? If not, why?
- [X] End-to-End tests have been added or updated?
- [ ] Was the documentation updated, or a dedicated issue for documentation created? (If applicable)
- [ ] Is the release notes updated? (If applicable)

### CODE REVIEW

File: tests/gui/e2e/test_resizing.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@pytest.mark.teste2e
+def test_has_default_value(page: Page, gui: Gui, helpers):
+    percentages = [
+        (1852, 50.83),
+        (1856, 45.29),
+        (1860, 39.65),
+        (1864, 55.03),
+    ]
+    data = pandas.DataFrame(percentages, columns=["Year", "%"]) # noqa: F841
+    page_md = """
```
Comment (FabienLelaquais) [inline tests/gui/e2e/test_resizing.py]:
```suggestion
    page_md = "<|{data}|chart|type=bar|x=Year|y=%|>"
```

File: tests/gui/e2e/test_resizing.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    page.wait_for_selector(".plot-container")
+    page.set_viewport_size({"width": 800, "height": 600})
+    elements = page.locator(
+        'path[style*="vector-effect: non-scaling-stroke; opacity: 1; stroke-width: 0px; fill: rgb(99, 110, 250); fill-opacity: 1;"]') # noqa: E501
+    first_element = elements.first
+    box_before = first_element.bounding_box()
+    page.set_viewport_size({"width": 1920, "height": 1080})
+    page.wait_for_timeout(1000)
+    box_after = first_element.bounding_box()
+    assert box_after['width'] > box_before['width']
```
Comment (FabienLelaquais) [inline tests/gui/e2e/test_resizing.py]:
```suggestion
    assert box_after["width"] > box_before["width"]
```

File: taipy/gui/viselements.json [hunk 3]
```diff
@@ -1427,14 +1427,14 @@
                     {
                         "name": "width",
                         "type": "Union[str,number]",
-                        "default_value": "None",
-                        "doc": "The width of the metric control, in CSS units."
+                        "default_value": "20vw",
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]:
```suggestion
                        "default_value": "\"20vw\"",
```

File: taipy/gui/viselements.json [hunk 4]
```diff
@@ -1427,14 +1427,14 @@
...
+                        "doc": "The width of the metric control, in CSS units"
                     },
                     {
                         "name": "height",
                         "type": "Union[str,number]",
-                        "default_value": "None",
-                        "doc": "The height of the metric control, in CSS units."
+                        "default_value": "20vh",
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]:
```suggestion
                        "default_value": "\"20vh\"",
```

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 5]
```diff
@@ -52,19 +52,30 @@ interface MetricProps extends TaipyBaseProps, TaipyHoverProps {
     template_Light_?: string;
 }
 
-const emptyLayout = {} as Partial<Layout>;
-const defaultStyle = { position: "relative", display: "inline-block", width: "100%" } as CSSProperties;
+const defaultLayout = { margin: { l: 50, r: 50, t: 50, b: 50 } } as Partial<Layout>;
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx:55]: Why such large margins?

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 6]
```diff
@@ -52,19 +52,30 @@ interface MetricProps extends TaipyBaseProps, TaipyHoverProps {
...
+const defaultStyle = {
+    position: "relative",
+    display: "inline-block",
+    width: "20vw",
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx:62]: May be a little comment here and next line indicating this impact viselements.json - in order to remember in the future

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Looking good.
A few details though

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx (hunk 5 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:55]: If the margin is set to 0, the metric’s number overflows outside the container, making it invisible. Adjusting it as shown ensures a better appearance when the default width and height are 20vw and 20vh, respectively. If the user want different margin then they can use the "layout" property to override it

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:55]: 👍

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**